pkg/commands/domainv1/tools/status.go

```diff
@@ -0,0 +1,95 @@
+package tools
+
+import (
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/fastly/cli/pkg/argparser"
+	fsterr "github.com/fastly/cli/pkg/errors"
+	"github.com/fastly/cli/pkg/global"
+	"github.com/fastly/go-fastly/v10/fastly"
+	"github.com/fastly/go-fastly/v10/fastly/domains/v1/tools/status"
+)
+
+// GetDomainStatusCommand calls the Fastly API to check the availability status of a domain.
+type GetDomainStatusCommand struct {
+	argparser.Base
+	argparser.JSONOutput
+	// Required.
+	domain string
+	// Optional.
+	scope argparser.OptionalString
+}
+
+// NewDomainStatusCommand returns a usable DomainStatusCommand registered under the parent.
+func NewDomainStatusCommand(parent argparser.Registerer, g *global.Data) *GetDomainStatusCommand {
+	cmd := GetDomainStatusCommand{
+		Base: argparser.Base{
+			Globals: g,
+		},
+	}
+
+	cmd.CmdClause = parent.Command("status", "Check the availability status of a single domain name.")
+	// Required.
+	cmd.CmdClause.Flag("domain", "Domain being checked for availability").Required().StringVar(&cmd.domain)
+	// Optional.
+	cmd.CmdClause.Flag("scope", "Scope determines the availability check to perform, specify `estimate` for an estimated check").Action(cmd.scope.Set).StringVar(&cmd.scope.Value)
+	cmd.RegisterFlagBool(cmd.JSONFlag())
+
+	return &cmd
+}
+
+// Exec invokes the application logic for the command.
+func (g *GetDomainStatusCommand) Exec(_ io.Reader, out io.Writer) error {
+	if g.Globals.Verbose() && g.JSONOutput.Enabled {
+		return fsterr.ErrInvalidVerboseJSONCombo
+	}
+
+	fc, ok := g.Globals.APIClient.(*fastly.Client)
+	if !ok {
+		return errors.New("failed to convert interface to a fastly client")
+	}
+
+	input := &status.GetInput{
+		Domain: g.domain,
+	}
+
+	if g.scope.WasSet {
+		input.Scope = fastly.ToPointer(status.Scope(g.scope.Value))
```

Comment:
Should we validate the scope provided against `status.ScopeEstimate`?

e.g. what happens when `g.scope.Value` is set to something like "nonsense"?

I guess the API call would just fail, but as we have `status.ScopeEstimate` defined in go-fastly I wonder if we should use it or not.

Reply:
Yeah good point, I think we should probably throw an error? It makes sense to actually validate, if this is not set correctly, the API will just assume that a non-estimate scope was set. A response will still be returned without an estimated scope.

Reply:
```
	if g.scope.WasSet {
		scope := status.Scope(g.scope.Value)
		if scope != status.ScopeEstimate {
			return fsterr.RemediationError{
				Inner:       errors.New("invalid scope provided"),
				Remediation: "Use --scope estimate for an estimate check",
			}
		}
		input.Scope = fastly.ToPointer(scope)
	}
```
I believe this logic should cover that scenario and prompt a helpful error message

```
ERROR: invalid scope provided.

Use --scope estimate for an estimate check
```

Reply:
Updated this to use backticks for better error formatting: `Use `--scope=estimate` for an estimate check`